File: src/main/java/org/droidmate/analyzer/wrappers/BoxMateConsts.java
package org.droidmate.analyzer.wrappers;

import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

/**
 * Constants used by BoxMate
 */
public class BoxMateConsts {
    static final String ARGS_DIR = "-apksDir=%s";
    static final String ARGS_INLINE = "-inline";
    static final String ARGS_API23 = "-apiLevel=23";
    static final String ARGS_RESET = "-resetEvery=30";
    static final String ARGS_TIME = "-timeLimit=900";
    static final String ARGS_SEED = "-randomSeed=0";
    static final String ARGS_DEVICE = "-device=%d";
    static final String ARGS_EXPL_OUTPUT_DIR = "-outputDir=%s";
    static final String ARGS_SNAP = "-getValidGuiSnapshotRetryAttempts=2";
    static final String ARGS_REPLACE_RESOURCES = "-replaceExtractedResources=false";
    static final String ARGS_UNPACK = "-unpack";

    public static final String FILE_API_POLICIES = "api_policies.txt";
}

File: src/main/kotlin/org/droidmate/analyzer/wrappers/BoxMateConsts.kt
package org.droidmate.analyzer.wrappers

/**
 * Constants used by BoxMate
 */
object BoxMateConsts {
    internal val ARGS_DIR = "-apksDir=%s"
    internal val ARGS_INLINE = "-inline"
    internal val ARGS_API23 = "-apiLevel=23"
    internal val ARGS_RESET = "-resetEvery=30"
    internal val ARGS_TIME = "-timeLimit=900"
    internal val ARGS_SEED = "-randomSeed=0"
    internal val ARGS_DEVICE_SEQ = "-device=%d"
    internal val ARGS_DEVICE_SN = "-deviceSN=%s"
    internal val ARGS_EXPL_OUTPUT_DIR = "-outputDir=%s"
    internal val ARGS_SNAP = "-getValidGuiSnapshotRetryAttempts=2"
    internal val ARGS_REPLACE_RESOURCES = "-replaceExtractedResources=false"
    internal val ARGS_LAUNCH_ACTIVITY_DELAY = "-launchActivityDelay=20000"

    val FILE_API_POLICIES = "api_policies.txt"
}
